### PR TITLE
fix sync memory policy lose effect when inserting too many partitions…

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #25763

### CODE REVIEW

File: internal/datanode/channel_meta.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ type Channel interface {
 
 	updateSegmentRowNumber(segID UniqueID, numRows int64)
 	updateSegmentMemorySize(segID UniqueID, memorySize int64)
+	updateSingleSegmentMemorySize(segID UniqueID)
```
Comment (bigsheeper) [inline internal/datanode/channel_meta.go:87]: we'd better to modify updateSegmentMemorySize so we don't have to add new api.

**Comment by MrPresent-Han** [comment]
rerun ut

File: internal/datanode/channel_meta.go [hunk 2]
```diff
@@ -601,11 +605,21 @@ func (c *ChannelMeta) updateSegmentRowNumber(segID UniqueID, numRows int64) {
...
+	return memorySize
+}
 
+func (c *ChannelMeta) updateSingleSegmentMemorySize(segID UniqueID) {
+	c.segMu.RLock()
+	defer c.segMu.RUnlock()
+	memorySize := c.calculateSegmentMemorySize(segID)
 	log.Info("updating segment memorySize", zap.Int64("segmentID", segID), zap.Int64("memorySize", memorySize))
 	seg, ok := c.segments[segID]
 	if ok && seg.notFlushed() {
```
Comment (xiaofan-luan) [inline internal/datanode/channel_meta.go:625]: if segment flushed, will this update working?

**Pull request closed without merging**